avocado/plugins/diff.py

```diff
@@ -0,0 +1,421 @@
+                                            str(test['status']))
+                results2.append(test_result)
+
+            if str(results1) != str(results2):
+                test_results_header = ['\n',
+                                       '# TEST RESULTS\n']
+                job1_results.extend(test_results_header)
+                job1_results.extend(results1)
+                job2_results.extend(test_results_header)
+                job2_results.extend(results2)
+
+        if 'all' in args.diff_filter or 'config' in args.diff_filter:
+            config1 = self._get_config(job1_dir)
+            config2 = self._get_config(job2_dir)
+
+            if str(config1) != str(config2):
+                config_header = ['\n',
+                                 '# SETTINGS\n']
+                job1_results.extend(config_header)
+                job1_results.extend(config1)
+                job2_results.extend(config_header)
+                job2_results.extend(config2)
+
+        if 'all' in args.diff_filter or 'sysinfo' in args.diff_filter:
+            sysinfo_pre1 = self._get_sysinfo(job1_dir, 'pre')
+            sysinfo_pre2 = self._get_sysinfo(job2_dir, 'pre')
+
+            if str(sysinfo_pre1) != str(sysinfo_pre2):
+                sysinfo_header_pre = ['\n',
+                                      '# SYSINFO PRE\n']
+                job1_results.extend(sysinfo_header_pre)
+                job1_results.extend(sysinfo_pre1)
+                job2_results.extend(sysinfo_header_pre)
+                job2_results.extend(sysinfo_pre2)
+
+            sysinfo_post1 = self._get_sysinfo(job1_dir, 'post')
+            sysinfo_post2 = self._get_sysinfo(job2_dir, 'post')
+
+            if str(sysinfo_post1) != str(sysinfo_post2):
+                sysinfo_header_post = ['\n',
+                                       '# SYSINFO POST\n']
+                job1_results.extend(sysinfo_header_post)
+                job1_results.extend(sysinfo_post1)
+                job2_results.extend(sysinfo_header_post)
+                job2_results.extend(sysinfo_post2)
+
+        if getattr(args, 'create_reports', False):
+            self.std_diff_output = False
+            prefix = 'avocado_diff_%s_' % job1_id[:7]
+            tmp_file1 = tempfile.NamedTemporaryFile(prefix=prefix,
+                                                    suffix='.txt',
+                                                    delete=False)
+            tmp_file1.writelines(job1_results)
+            tmp_file1.close()
+
+            prefix = 'avocado_diff_%s_' % job2_id[:7]
+            tmp_file2 = tempfile.NamedTemporaryFile(prefix=prefix,
+                                                    suffix='.txt',
+                                                    delete=False)
+            tmp_file2.writelines(job2_results)
+            tmp_file2.close()
+
+            msg = '%s %s' % (tmp_file1.name, tmp_file2.name)
+            LOG.info(msg)
```

Comment:
nitpick: LOG accepts arguments, you can simply do `LOG.debug("%s %s", tmp_file1.name, tmp_file2.name)` :wink: